test/integration/smoke.test.ts

```diff
@@ -150,50 +150,33 @@ describe('API Client Integration Tests', () => {
   });
 });
 
-async function waitForReplication(
-  api: XataApiClient,
-  workspace: string,
-  database?: string,
-  iteration = 0
-): Promise<void> {
+async function waitForReplication(api: XataApiClient, workspace: string, database?: string): Promise<void> {
   try {
     if (database === undefined) {
       await api.database.getDatabaseList({ workspace });
     } else {
       await api.branches.getBranchList({ workspace, database, region });
     }
   } catch (error) {
-    console.log(
-      `Waiting for create ${database === undefined ? 'API key' : 'database'} replication to finish... (${iteration})`
-    );
+    console.log(`Waiting for create ${database === undefined ? 'API key' : 'database'} replication to finish...`);
     await new Promise((resolve) => setTimeout(resolve, 2000));
-    return await waitForReplication(api, workspace, database, iteration + 1);
+    return await waitForReplication(api, workspace, database);
```

Comment:
Is/was this going to loop forever if replication fails?

Reply:
We have a timeout at 2 minutes that stops any loop.

Reply:
Nice, TY!